cinema/tests/test_movie_viewset.py

```diff
@@ -0,0 +1,256 @@
+    return MovieSession.objects.create(**defaults)
+
+
+def image_upload_url(movie_id):
+    """Return URL for image upload"""
+    return reverse("cinema:movie-upload-image", args=[movie_id])
+
+
+def detail_url(movie_id):
+    return reverse("cinema:movie-detail", args=[movie_id])
+
+
+class UnauthenticatedMovieApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_auth_required(self):
+        res = self.client.get(MOVIE_URL)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class AuthenticatedMovieApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="test_user1@test.com",
+            password="test_password1",
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_movie_list(self):
+        sample_movie()
+        movie_with_genres = sample_movie()
+        movie_with_actors = sample_movie()
+
+        genre = sample_genre()
+        actor = sample_actor()
+
+        movie_with_genres.genres.add(genre)
+        movie_with_actors.actors.add(actor)
+
+        response = self.client.get(MOVIE_URL)
+        movies = Movie.objects.all()
+        serializer = MovieListSerializer(movies, many=True)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serializer.data)
+
+    def test_filter_movies_by_title(self):
+        movie_1 = sample_movie(title="Fury")
+        movie_2 = sample_movie(title="Fast & Furious")
+
+        response = self.client.get(MOVIE_URL, {"title": "Fur"})
+
+        serializer_1 = MovieListSerializer(movie_1)
+        serializer_2 = MovieListSerializer(movie_2)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn(serializer_1.data, response.data)
+        self.assertIn(serializer_2.data, response.data)
+
+    def test_filter_movies_by_genre(self):
+        genre_1 = sample_genre(name="Action")
+        genre_2 = sample_genre(name="Horror")
+
+        movie = sample_movie()
+        movie_action = sample_movie()
+        movie_action_horror = sample_movie()
+
+        movie_action.genres.add(genre_1)
+        movie_action_horror.genres.add(genre_1, genre_2)
+
+        response = self.client.get(
+            MOVIE_URL,
+            {"genres": f"{genre_1.id},{genre_2.id}"}
+        )
+
+        serializer = MovieListSerializer(movie)
+        serializer_action = MovieListSerializer(movie_action)
+        serializer_action_horror = MovieListSerializer(movie_action_horror)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn(serializer_action.data, response.data)
+        self.assertIn(serializer_action_horror.data, response.data)
+        self.assertNotIn(serializer.data, response.data)
+
+    def test_filter_movies_by_actor(self):
+        actor_1 = sample_actor()
+        actor_2 = sample_actor()

```

Comment:
In `test_filter_movies_by_actor`, both `actor_1` and `actor_2` are created without specifying unique attributes, which might lead to them being identical. Consider specifying unique attributes to ensure they are distinct actors.